### PR TITLE
TableSchema check improvements

### PR DESCRIPTION
**Pull request by mhidas**
Trying to improve and slightly extend the functionality for this check type to support the Animal Tracking use case.

### CODE REVIEW

**Review by digorgonzola — APPROVED** [review]

File: test_aodncore/pipeline/steps/test_check.py [hunk 1]
```diff
@@ -223,3 +235,16 @@ def test_missing_schema_file(self):
...
+                                           check_params={"tableschema_filename_pattern": "[^_]*(_[^_]*)?"}
+                                           )
+        collection = PipelineFileCollection(SAME_SCHEMA_CSVS)
+        ts_runner.run(collection)
+
+        failed_files = [(f.name, f.check_result.log)
+                        for f in collection
+                        if not f.check_result.compliant
+                        ]
+        self.assertEqual([], failed_files)
```
Comment (alexgleith) [inline test_aodncore/pipeline/steps/test_check.py:250]: You could do a length check, instead of comparing two empty tuples.

**Review by alexgleith — APPROVED** [review]
Looks good to me.

One improvement might be to add a test that finds a file that doesn't match a pattern, so test for a failed check.

File: aodncore/pipeline/schema.py [hunk 2]
```diff
@@ -22,7 +22,8 @@
         'criteria': {'type': 'string'},
         'skip_checks': {'type': 'array', 'items': {'type': 'string'}},
         'output_format': {'type': 'string'},
-        'verbosity': {'type': 'integer'}
+        'verbosity': {'type': 'integer'},
+        'tableschema_filename_pattern': {'type': 'array'}
```
Comment (evacougnon) [inline aodncore/pipeline/schema.py:26]:
```suggestion
        'tableschema_filename_pattern': {'type': 'string'}
```

File: test_aodncore/pipeline/steps/test_check.py (hunk 1 above)
Comment (mhidas) [reply test_aodncore/pipeline/steps/test_check.py]: :+1: Actually I didn't even need to create a list (it was just helpful while debugging)

**Comment by mhidas** [comment]
@digorgonzola @nspool This is ready to merge, if you have a few minutes for a final look. I've added a bit more to the unittests after suggestions from @alexgleith .

**Comment by nspool** [comment]
When I run the tests locally the test `test_compliant_file` is failing for me with the warning:
```
DeprecationWarning: Converting `np.character` to a dtype is deprecated. The current result is `np.dtype(np.str_)` which is not strictly correct. Note that `np.character` is generally deprecated and 'S1' should be used.
```

Running Python 3.8.16

**Comment by nspool** [comment]
@mhidas after running `pip install -r test_requirements.txt` these versions are installed:

```
compliance-checker        4.1.1
numpy                     1.24.2
```

Trying `pip install compliance-checker==5.0.1` fails as that version is not available, only `5.0.0` or `5.0.2`.

**Comment by nspool** [comment]
Thanks @mhidas , I rebased from master locally and all the tests pass. Happy to merge.

**Review by nspool — APPROVED** [review]

**Comment by mhidas** [comment]
:+1: Thanks @nspool

**Pull request merged**